lib/util.c

```diff
@@ -712,3 +713,24 @@ fast_search_free(fast_search_t *self)
 extern inline size_t sub_idx_1st_strict_upper_bound(
     const double *base, size_t start, size_t stop, double query);
 extern inline size_t fast_search_idx_strict_upper(fast_search_t *self, double query);
+
+/*   The gsl_ran_flat() function is supposed to output lo<=x<hi, but
+ *   sometimes (with probability <1e-9) we find x=hi.
+ *   This function includes a while loop to ensure x<hi.
+ *   https://github.com/tskit-dev/msprime/issues/1997
+ */
+double
+msp_gsl_ran_flat(gsl_rng *rng, double lo, double hi)
+{
+    double x;
+
+    if (lo > hi) {
+        double temp = lo;
```

Comment:
Style nitpick: we do all the declarations at the start of the function, so here it'd be ``double x, temp;``.

No particularly good reason for it anymore tbh, but best to stick with the established style at this point.